### DISCUSSION TITLE
Shape connector lines through other shapes

### DISCUSSION THREAD

**Discussion by shockogit** [jgraph/drawio-desktop#1484] [Q&A]
I'm new to draw.io but I have a simple use case. I have the following diagram:

![image](https://github.com/jgraph/drawio-desktop/assets/46563321/e78d709a-e1fb-45fd-b5d5-b7f66ec53d5b)

I have simply drawn a connector from A -> C. Now ifI move A or C the connector remains atatched to both but I also wish it to pass **_through_** B at all times when I do this it naturally simlpy continues to connect A -> C

![image](https://github.com/jgraph/drawio-desktop/assets/46563321/21ce75bb-b264-445c-b251-d7f93867096c)

How can I get this connection from A -> to always flow through B and ideally as a single connector/line (so it maintains any formatting I put on it?

The closest I have been able to get to it is locking B and adding a waypoint to the connect from A -> C

![image](https://github.com/jgraph/drawio-desktop/assets/46563321/68eda8f5-382d-434e-9ddb-45d81b48f4fa)

**Comment by shockogit** [comment]
Anyone?

**Reply by gammalogic** [reply]
I did this by creating a waypoint in the center of box "B", then clicking and dragging on the screen to highlight the two elements together to group them (not lock them). Then, create two arrows that connect boxes "A" and "C" to the waypoint. Then, when you move "B" the connections update as expected. You then just need to use the "Arrange" menu to reposition the waypoint behind "B" to hide it.

![Grouping_Example drawio](https://github.com/jgraph/drawio-desktop/assets/28597602/be67ef55-9938-4f19-b1a8-2c1d359c8d04)

**Comment by gammalogic** [comment]
Just to add, this arrangement does work but the connecting lines do not always reorient themselves as might be expected because of the waypoint being centered exactly in the middle of "B" and the connections to "A" and "C" being fixed to one side of the element, so it might not always look nicely drawn out. This can be corrected by moving the arrow "handles" to a better position, but it is not done automatically.
